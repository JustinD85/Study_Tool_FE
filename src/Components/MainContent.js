import React, { Component } from 'react';
import Game from './Game';
import Training from './Training';
import Menu from './Menu';
import Profile from './Profile';

class MainContent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      status: 'profile',
    }
  }

  handleContentView = (status) => {

    this.setState({
      status
    });
  }

  userStatus = () => {
    const { status } = this.state;
    const userStatus = {
      game: <Game info={{ ...this.props }} />,
      train: <Training info={{ ...this.props }} />,
      profile: <Profile info={{ ...this.props }} />
    }

    return userStatus[status];
  }

  render() {
    const { user, handleAppView } = this.props;
    const { handleContentView } = this;

    return (
      <div className="MainContent">
        <Menu
          handleContentView={handleContentView}
          handleAppView={handleAppView}
          user={user}
          currentView={this.state.status}
        />
        {this.userStatus()}
      </div>
    );
  }
}

export default MainContent;
