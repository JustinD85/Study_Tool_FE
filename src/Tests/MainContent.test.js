import React from 'react';
import MainContent from '../Components/MainContent';
import { shallow } from 'enzyme';

describe('<MainContent/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MainContent />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});