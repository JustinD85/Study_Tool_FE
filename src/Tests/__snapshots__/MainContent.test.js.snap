// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContent/> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Menu
          currentView="profile"
          handleContentView={[Function]}
        />,
        <Profile
          info={Object {}}
        />,
      ],
      "className": "MainContent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currentView": "profile",
          "handleAppView": undefined,
          "handleContentView": [Function],
          "user": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "info": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Menu
            currentView="profile"
            handleContentView={[Function]}
          />,
          <Profile
            info={Object {}}
          />,
        ],
        "className": "MainContent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentView": "profile",
            "handleAppView": undefined,
            "handleContentView": [Function],
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "info": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
